### ISSUE TITLE
Minimum PHP version in README file

### ISSUE THREAD

**Issue by progga** [jsdrupal/drupal-admin-ui#317]
Hello,
I was following the instructions in the README to setup drupal-admin-ui locally.  I got told off by composer almost as soon as I got started :(  The root of the trouble is that I am on Debian Stable (AKA Stretch) which is carrying PHP 7.0.x and [doctrine/annotations](https://packagist.org/packages/doctrine/annotations) et al now demands **php ^7.1**.  The otherwise excellent README mentions **PHP 5.5.9 or greater** as a requirement and this seem to have become outdated with the advent of Drupal 8.7.x.  So I thought I will update the README accordingly.  Pull request to follow.

As an aside, I had to summon the help of Docker to get me out of trouble.  After tinkering around for a day or so, I eventually managed to reach the promised land.  I have left the [Dockerfile as a gist](https://gist.github.com/progga/37bb352c18af46ba2489fed6414e1b15) in case anyone finds it useful.

**Comment by dawehner** [comment]
I think we should limit the version of the doctrine library to be as stricter. IMHO following core's PHP version makes sense. What do you think about this alternative proposal?

**Comment by progga** [comment]
Hello @dawehner 

> I think we should limit the version of the doctrine library to be as stricter.

If we want to take this route, we can **drop** jsdrupal/drupal-admin-ui-demo's composer.lock file.  [drupal-composer/drupal-project](https://github.com/drupal-composer/drupal-project) seem to have fathered jsdrupal/drupal-admin-ui-demo and that one does **not** commit its composer.lock file.  Perhaps we shouldn't either.  When composer.lock is absent, composer will install whatever version is appropriate.

> IMHO following core's PHP version makes sense.

Totally makes sense :)

## See also
- https://www.drupal.org/project/drupal/issues/2843328

**Comment by dawehner** [comment]
Well, an alternative would be setting the requirement on doctrine to ensure this doesn't happen.

I think especially for having a stable environment for testing etc. having a composer.lock is useful.

Note: drupal project doesn't ship with one because they are just a template for your own site.

**Comment by progga** [comment]
> an alternative would be setting the requirement on doctrine to ensure this doesn't happen.

Okay, to go down this way we will have to pin the following 12 packages to their corresponding versions from [drupal/drupal's composer.lock](https://github.com/drupal/drupal/blob/8.6.x/composer.lock):
- doctrine/annotations: ~1.2.7
- doctrine/cache: ~1.6.1
- doctrine/collections: ~1.3.0
- doctrine/common: ~2.6.2
- doctrine/event-manager: ?
- doctrine/inflector: ~1.1.0
- doctrine/persistence: ?
- doctrine/reflection: ?
- doctrine/instantiator: ~1.0.5
- myclabs/deep-copy: ?
- symfony/browser-kit: ~3.4.14
- symfony/dom-crawler: ~3.4.14

I can think of only one potential problem with this approach which is that, if drupal/core adds another package that needs pinning, it will break jsdrupal/drupal-admin-ui-demo until someone pins that package manually.

Given that setting [config.platform.php to 5.5.9 in composer.json is no longer advised](https://www.drupal.org/project/drupal/issues/2843328), another option will be to generate the composer.lock file under a real PHP 5.5.9 environment using Ubuntu 14.04

@dawehner Let me know which one of the above two solutions you prefer and I am happy to give it a go :)

> drupal project doesn't ship with one because they are just a template for your own site.

Got it.  It is indeed described as a "Composer template for Drupal projects".

**Comment by dawehner** [comment]
I'm wondering what others like @justafish @lauriii think about this problem.

**Comment by progga** [comment]
Just to summarize, here's what we need to decide:
- Given that Drupal 8's minimum PHP requirement is 5.5.9, should jsdrupal/drupal-admin-ui support it or **stick** to PHP 7.1+?  The [PHP requirement page](https://www.drupal.org/docs/8/system-requirements/php-requirements#php_recommended) also says
> ... We recommend updating to at least PHP 7.1 if possible, and ideally PHP 7.2, ...
So there are cases both for and against.
- If we want to support PHP 5.5.9, how do we produce the composer.lock file?  Should we use a PHP 5.5.9 environment to generate it or should we pin packages to versions acceptable to PHP 5.5.9?

**Comment by lauriii** [comment]
> IMHO following core's PHP version makes sense

I don't think it makes sense to support PHP 5.x anymore since support for them is[ dropped from Drupal core in the next release](https://www.drupal.org/node/2938726). I think it's fine if we only support PHP 7.0 or higher.

**Comment by dawehner** [comment]
That is a strong point!

**Comment by dawehner** [comment]
Sadly it doesn't help with the ```doctrine/annotations``` problem, given it requires php 7.1.

**Comment by progga** [comment]
I am assuming we have been unable to reach a conclusion regarding the minimum PHP version.  I will maintain status quo and close this ticket shortly.

**Comment by dawehner** [comment]
Wasn't our conclusion to support 7.0.x?

**Comment by progga** [comment]
Thanks for the clarification @dawehner .  In that case I will raise another pull request with the following changes:
- Updated PHP version in the README file.
- A composer.**lock** file **regenerated** under PHP 7.0.8

Does that sound okay?

**Comment by dawehner** [comment]
@progga Seems reasonable :)

**Comment by progga** [comment]
Thanks @dawehner, I will come up with the above changes within a few days.
